File: csharp/log-levels/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine)
        => logLine.Split(':')[1].Trim();


    public static string LogLevel(string logLine)
    {
        // char position after the leading left bracket '['
        var startIndex = logLine.IndexOf('[') + 1;
        // char position before the closing right bracket ']'
        var endIndex = logLine.IndexOf(']') - 1;
        // take a substring from startIndex to endIndex and lowercase it
        return logLine.Substring(startIndex, endIndex).ToLower();
    }

    public static string Reformat(string logLine)
        => $"{Message(logLine)} ({LogLevel(logLine)})";
}

File: java/log-levels/src/main/java/LogLevels.java
public class LogLevels {

    // message returns the message of the logline with the loglevel removed.
    public static String message(String logLine) {
        return logLine.split(":", 2)[1].trim();
    }

    // logLevel returns the log level in lowercase.
    public static String logLevel(String logLine) {
        return logLine.substring(1, logLine.indexOf("]")).toLowerCase();
    }

    // reformat returns the logLine in a new format.
    public static String reformat(String logLine) {
        return message(logLine) + " (" + logLevel(logLine) + ")";
    }
}
